creation/views.py

```diff
@@ -2861,3 +2943,567 @@ def update_codefiles(request):
+                language_id=22).order_by(
+                'tutorial_detail__foss__foss','language__name','tutorial_detail__order')
+            bid_count = ContributorRole.objects.filter(user_id = user.id ,status=1).exclude(language_id=22).aggregate(Count('id'))
+       
+        
+    else:
+        raise PermissionDenied()
+    
+    extension = []
+    pub_tutorials_set = final_query
+    context['datetoday'] = datetime.now()
+    raw_get_data = request.POST.get('o', None)
+    tutorials_sorted = get_sorted_list(request, pub_tutorials_set, header, raw_get_data)
+    
+    ordering = get_field_index(raw_get_data)
+    tutorials = CreationStatisticsFilter(request.POST, queryset=tutorials_sorted)
+
+    
+    if bid_count != 0:
+        context['tutorials_count'] = bid_count['id__count'] + tutorials.qs.aggregate(Count('id'))['id__count']    
+    try:
+        context['bid_count__count'] = bid_count['id__count']
+        if sel_status == 'ongoing':
+            context['perc'] =  bid_count['id__count'] * 100 / (bid_count['id__count'] + tutorials.qs.aggregate(Count('id'))['id__count'])
+        else:
+            context['perc'] = float(bid_count['id__count'] *100) / float(tutorials.qs.aggregate(Count('id'))['id__count'] + bid_count['id__count'])
+    except:
+        context['bid_count__count'] = 0
+        context['perc'] = 0
+    
+    form = tutorials.form
+    if lang_qs: 
+        form.fields['language'].queryset = lang_qs
+
+
+    if contributors_list:
+            form.fields['script_user'].queryset = contributors_list       
+    
+    context['form'] = form
+        
+    # Pagination
+    paginator = Paginator(tutorials, 50)
+    try:
+        page = int(request.GET.get('page','1'))
+    except:
+        page = int(request.GET.get('page'))
+
+    try:
+        posts = paginator.page(page)
+        
+    except(EmptyPage,PageNotAnInteger):
+        posts = paginator.page(paginator.num_pages)
+    context['collection'] = posts
+    context['header'] = header
+    context['ordering'] = ordering
+    context['status'] = active
+    context['counter'] = itertools.count(1)
+
+    if request.method == 'POST':
+        language_id = request.POST.get('language')
+        if language_id:
+            contributors_updated = RoleRequest.objects.filter(role_type=0,status = 1,language=language_id).values_list('user__id').distinct();
+            rated_contributors = ContributorRating.objects.filter(language_id=language_id,user_id__in=contributors_updated).values_list('user_id','user__username','rating')
+            context['contributors'] = rated_contributors
+    
+    context.update(csrf(request))
+    if is_language_manager(request.user):
+        return render(request, 'creation/templates/allocate_tutorial_manager.html', context)
+    else:
+        return render(request, 'creation/templates/allocate_tutorial.html', context)
+
+#global variable
+submissiondate = datetime.date(datetime.now())
+
+def refresh_tutorials(request):
```

Comment:
This view is neither checking authentication nor the type of user!
Also, is there a need of a separate view function to simply refresh?
Rethink on this.